### PR TITLE
Dawn: Support DoClear (by queueing the work)

### PR DESCRIPTION
**Pull request by dneto0**

### CODE REVIEW

File: src/dawn/engine_dawn_test.cc [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+namespace dawn {
+
+namespace {
+
+using EngineDawnTest = testing::Test;
+
+TEST_F(EngineDawnTest, PendingWorkDefaultIsEmpty) {
+  EngineDawn e;
+  auto& pending = e.GetPendingWorkForTest();
+  EXPECT_EQ(0, pending.size());
```
Comment (dj2) [inline src/dawn/engine_dawn_test.cc]: EXPECT_TRUE(pending.empty());

File: src/dawn/engine_dawn_test.cc [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "src/dawn/engine_dawn.h"
+
+#include "gmock/gmock.h"
```
Comment (dj2) [inline src/dawn/engine_dawn_test.cc]: gtest/gtest.h? Since you aren't mocking anything?

Comment (dneto0) [reply src/dawn/engine_dawn_test.cc]: Will fix.  I reflexively use gmock.h because I usually use EXPECT_THAT with the more powerful predicates.

**Comment by dneto0** [comment]
I'm changing strategy:  Instead of queueing work, I'm going to execute them immediately (actually by adding to the command buffer)

**Pull request merged**